rest/src/main.rs

```diff
@@ -754,6 +782,181 @@ fn ibc_applications_transfer_params() -> Value {
+            }
+        }
+    })
+}
+
+#[get("/cosmos/base/tendermint/v1beta1/blocks/<height>")]
+async fn block(height: u32) -> Value {
+    let client = tm::HttpClient::new(app_host()).unwrap();
+
+    let res = client
+        .block(tendermint::block::Height::from(height))
+        .await
+        .unwrap();
+
+    let last_commit = res.block.last_commit.unwrap();
+    let signatures: Vec<_> = last_commit.signatures
+        .iter()
+        .map(|signature| -> Value {
+            let signature_raw = RawCommitSig::from(signature.clone());
+
+            json!({
+                "validator_address": base64::encode(signature_raw.validator_address),
+                "block_id_flag": signature_raw.block_id_flag,
+                "timestamp": signature_raw.timestamp,
+                "signature": base64::encode(signature_raw.signature),
+            })
+        })
+        .collect();
+
+    json!({
+        "block_id": res.block_id,
+        "block": {
+            "header": res.block.header,
+            "data": res.block.data,
+            "evidence": res.block.evidence,
+            "last_commit": {
+                "block_id": last_commit.block_id,
+                "signatures": signatures
+            }
+        }
+    })
+}
+
+#[get("/cosmos/base/tendermint/v1beta1/validatorsets/latest")]
+async fn latest_validator_set() -> Value {
+    let client = tm::HttpClient::new(app_host()).unwrap();
+
+    let block = client
+        .latest_block()
+        .await
+        .unwrap();
+
+    let res = client
+        .validators(block.block.header.height, tendermint_rpc::Paging::All)
+        .await
+        .unwrap();
+
+    let validators: Vec<_> = res.validators.iter()
+        .map(|validator| -> Value {
+            json!({
+                "address": validator.address,
+                "voting_power": (i64::from(validator.power) / 1_000_000).to_string(),
+                "proposer_priority": i64::from(validator.proposer_priority).to_string(),
+                "pub_key": {
+                    "@type": "/cosmos.crypto.ed25519.PubKey",
+                    "key": base64::encode(validator.pub_key.ed25519().unwrap().to_bytes()),
+                }
+            })
+        })
+        .collect();
+
+    json!({
+        "block_height": res.block_height,
+        "validators": validators,
+        "pagination": {
+            "next_key": null,
+            "total": res.validators.len(),
+        }
+    })
+}
+
+#[get("/cosmos/base/tendermint/v1beta1/validatorsets/<height>")]
+async fn validator_set(height: u32) -> Value {
+    let client = tm::HttpClient::new(app_host()).unwrap();
+
+    let res = client
+        .validators(height, tendermint_rpc::Paging::All)
+        .await
+        .unwrap();
+
+        let validators: Vec<_> = res.validators.iter()
+        .map(|validator| -> Value {
+            json!({
+                "address": validator.address,
+                "voting_power": (i64::from(validator.power) / 1_000_000).to_string(),
```

Comment:
actually I guess you are right, I added this because ping.pub (which I aimed at here) assumes this status is in nom rather than unom on staking page in 24h changes block:
<img width="1156" alt="image" src="https://github.com/nomic-io/nomic/assets/83376337/b399793a-6a73-4aa4-8692-748946a704c6">
if I remove this, it'll show values there that are 1 million bigger than this, but I guess this is a ping.pub issue and shouldn't be fixed this way, will revert this a bit later

Reply:
okay reverted